util/chplenv/chpl_comm.py

```diff
@@ -38,6 +38,11 @@ def get():
             else:
                 comm_val = 'none'
 
+    if comm_val == 'gasnet':
+        import chpl_platform
+        if get_network() == 'aries' or chpl_platform.get('target') == 'cray-xc':
+            error("CHPL_COMM=gasnet is no longer supported on Cray XC. Please use CHPL_COMM=ugni instead.")
```

Comment:
The latest GASNet snapshot removed aries-conduit, but other Cray XC platform support remains (albeit untested and thus subject to bit-rot).

Technically GASNet on Cray XC still supports both ofi-conduit and mpi-conduit, although neither was ever particularly performant there.